src/parquet/bloom_filter_algorithm.cc

```diff
@@ -0,0 +1,102 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "parquet/bloom_filter_algorithm.h"
+#include "arrow/status.h"
+#include "arrow/util/bit-util.h"
+#include "parquet/exception.h"
+#include "parquet/types.h"
+
+namespace parquet {
+constexpr uint32_t BlockBasedAlgorithm::SALT[BITS_SET_PER_BLOCK];
+
+BlockBasedAlgorithm::BlockBasedAlgorithm(uint32_t num_bytes)
+    : pool_(::arrow::default_memory_pool()) {
+  if ((num_bytes & (num_bytes - 1)) != 0) {
+    throw ParquetException("num_bytes is not a power of 2");
+  }
+
+  num_bytes_ = num_bytes;
+
+  ::arrow::Status status =
+      pool_->Allocate(num_bytes, reinterpret_cast<uint8_t**>(&bitset32_));
+  if (!status.ok()) {
```

Comment:
Use the `PARQUET_THROW_NOT_OK` macro to check the Status value

Reply:
Yes, thanks.